### DISCUSSION TITLE
ESP32-WROOM-32 and NeoGeo AES internal connection

### DISCUSSION THREAD

**Discussion by UnrealEditor1** [darthcloud/BlueRetro#144] [Q&A]
Hi,

I'm wondering if I can connect the ESP to the NeoGeo AES just without cable... technically straight at the controler output of the mainbiard.

I aslo have a big problem with the "cable building instruction". The pictures of the "BlueRetro LS 1.1" are different from the ESP32-WROOM-32. I don't understand how to connect it to the ESP because pinout description is different. Is there something I'm missing?

Regards

**Comment by darthcloud** [comment]
Only look at the DIY Through-hole section:
https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#diy-through-hole

You got the pdf schematic there, for internal install just ignore the DB25 and wire to ESP32 base on net name. You need the level shifter chips. ESP32 is 3.3v and NeoGeo is 5V.

**Reply by UnrealEditor1** [reply]
Thanks for the help! I tried to figure out how to use the schamatic and connect the DB25 to my ESP32. Would you mind to take a look at my picture and tell me if its correct!? I still don't know hoe to connect IO3 and I'm wondering if my ESP32 needs a level shifter becasue it has a 3V3 pin.

https://ibb.co/r4kBnRb

**Reply by darthcloud** [reply]
You cannot connect ESP32 directly to NeoGeo, you need the level shifter listed in https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-Cables-Build-Instructions#bill-of-materials

and connected like indicated in https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/NeoGeo.pdf just ignore DB25 and connect IOx name net to the ESP32.

**Reply by UnrealEditor1** [reply]
Hey DarthCloud,

thanks for taking your time! I'm very new to this and I try to do my best.

Is this correct? https://ibb.co/ZY0zzxK

I'm really not sure about the GND and Vcc part... I probably mess that up :(

**Reply by darthcloud** [reply]
Its all fine

**Reply by UnrealEditor1** [reply]
Would you mind helping me on the wiring diagram? Just for the ESP and the level shifter? Really don't want to destroy my NeoGeo AES. I can also pay you for this.

**Reply by darthcloud** [reply]
If you don't have the confidence to do this mod, don't do it.

Pay someone to do it. You will have find that someone yourself, I don't have any contacts for this.

And I'm not doing mod myself, I too busy with real family life.

Again I'm very sorry but can't help more than what the repo & wiki contain already.

**Reply by UnrealEditor1** [reply]
Hey there,

okay, no problem! Thanks for being honest and all the help so far! :)

**Reply by Adrian000111000** [reply]
> Hey DarthCloud,
> 
> thanks for taking your time! I'm very new to this and I try to do my best.
> 
> Is this correct? https://ibb.co/ZY0zzxK
> 
> I'm really not sure about the GND and Vcc part... I probably mess that up :(

WARNING! The connections in your is scheme are  99% OK but the correct IO3 port on WROOM PCB is the RXD 
DO NOT CONNECT ON SD3... I had Haomaru always jumping.

![Cattura](https://github.com/darthcloud/BlueRetro/assets/172551805/583ef2d6-b1b1-4eba-a810-406130f8be7f)

**Comment by pkerga** [comment]
hello
I received my level shifter and I start in the Neo geo.
I saw the diagram [above](https://ibb.co/ZY0zzxK) and was wondering if the esp32 needed a power supply ?
Nothing is plugged into the 3.3v or 5v of the esp32?

**Reply by darthcloud** [reply]
As a said earlier this schematic got all the info you need:
https://github.com/darthcloud/BlueRetroHW/blob/master/DIY/Para1P_OD.pdf

Ignore the DB25 and just connect things to the matching label name.

**Reply by pkerga** [reply]
Here where I am : 
[image](https://zupimages.net/up/22/09/kvr6.jpg)
I've only plugged in button A so far.
It still does not work.
Should i plug all buttons ?

PS: It works flawlessly on my DC!

**Reply by pkerga** [reply]
Got it !!!
I just need to learn to count correctly... I reversed the row from 8 to 14 on the 74AHCT125N...

**Reply by UnrealEditor1** [reply]
can you please make a diagram or good picture of the wireing?

**Comment by UnrealEditor1** [comment]
Hey there,Don’t use the diagram. I tried to find a question for same answer but nobody replied. All I know is you have to use 3x level switch which you can see in lower right corner but I don’t know how to connect level switch to esp. all I know is that you take power from Controller port to level switch…. Hope that somebody can continue or maybe change my diagram to finish it. Gesendet mit der mobilen Mail AppAm 02.03.22 um 10:51 schrieb pkerga
                
            
            
                
                    Von: "pkerga" ***@***.***>Datum: 2. März 2022An: "darthcloud/BlueRetro" ***@***.***>Cc: "Author" ***@***.***>,"UnrealEditor1" ***@***.***>Betreff: Re: [darthcloud/BlueRetro] ESP32-WROOM-32 and NeoGeo AES internal connection (Discussion #144)
                
                
hello
I received my level shifter and I start in the Neo geo.
I saw the diagram above and was wondering if the esp32 needed a power supply ?
Nothing is plugged into the 3.3v or 5v of the esp32?

**Reply by pmgducati** [reply]
The Diagram is correct, the 5VIN that comes from the controller port is a common connection between the 5V on the level shifters and the ESP32 (pin 7 on the DB25_Male Connector).  This is also true for all the GND connections (5x connections of the DB25_Male, Pins 2,5,7,9,12 on all three level shifters and on the controller plug). 

If you (@UnrealEditor1 / @pkerga) are interested, I am happy to design a PCB for this.  I already have 8 systems completed.  You could grab the gerber off my github and and have it made by your favorite CM (JLCPCB, Oshpark, PCBWay etc).

**Reply by pkerga** [reply]
I've never print a PCB. Maybe I should learn ?
Do you have pictures of the résults ?

**Reply by pmgducati** [reply]
When you order the PCBs they will come like this:
![PXL_20220302_221752797](https://user-images.githubusercontent.com/20894227/156459540-380d382a-2ae4-4c2d-9c47-62e1add94ad5.jpg)
![PXL_20220302_221757720](https://user-images.githubusercontent.com/20894227/156459544-2a33b445-e4eb-4388-baba-bf81980f4b0c.jpg)
![PXL_20220302_221800757](https://user-images.githubusercontent.com/20894227/156459548-e4363772-88aa-4244-904d-9cb425ca7f97.jpg)
![PXL_20220302_221801833](https://user-images.githubusercontent.com/20894227/156459550-6a616913-7f52-4781-ad53-e4080ed0730c.jpg)
![PXL_20220302_221803021](https://user-images.githubusercontent.com/20894227/156459552-a42b1dce-7b12-4a45-9cf2-914be1a55380.jpg)

This is what they look like Assembled (The BOM is in Github, basically the same as @darthcloud's Part Lists on the cables page. )
![PXL_20220302_221819353](https://user-images.githubusercontent.com/20894227/156459787-c2dccd88-0ee5-4ab6-a253-1fdb6b755baf.jpg)
![PXL_20220302_221823449](https://user-images.githubusercontent.com/20894227/156459791-79280da5-8c4e-4eac-868a-d4a23098f457.jpg)
![PXL_20220302_221828115](https://user-images.githubusercontent.com/20894227/156459795-fcb07774-ee1e-47ec-a796-a749eea0aa41.jpg)
![PXL_20220302_221839010](https://user-images.githubusercontent.com/20894227/156459799-53a21ec4-ed1d-4052-82dc-fa090c8fbfa8.jpg)
![PXL_20220302_221856953](https://user-images.githubusercontent.com/20894227/156459929-8dd72d50-5ef0-412e-a77e-fc6e42ab554e.jpg)

You are see the Main PCB, Gamecube, 3DO and PCEngine.

**Reply by pmgducati** [reply]
I am also working on a PCB with a alternative to the DB25 connector, I looking into a keyed cartridge slot, think something similar to the NES or Genesis Cart/Plug.

**Reply by darthcloud** [reply]
Cool stuff I'll add to the community links on main page.

**Reply by pmgducati** [reply]
Thanks!  I need to button up the comments and stuff!

**Reply by pmgducati** [reply]
@darthcloud https://github.com/pmgducati/Blue-Retro-AIO-Units
